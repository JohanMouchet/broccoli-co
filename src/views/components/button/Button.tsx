import React from "react";
import cx from "classnames";
import styles from "./Button.module.scss";

type Props = {
  type?: "button" | "submit";
  block?: boolean;
  onClick?: () => void;
  children?: React.ReactNode;
};

const Button: React.FC<Props> = ({
  type = "button",
  block,
  onClick,
  children,
}) => (
  <button
    className={cx(styles["button"], block && styles["block"])}
    type={type}
    onClick={onClick}
  >
    {children}
  </button>
);

export default Button;
