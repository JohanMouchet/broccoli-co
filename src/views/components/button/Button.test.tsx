import React from "react";
import { render } from "@testing-library/react";
import Button from ".";

it("Renders <Button /> without crashing", () => {
  expect(
    render(<Button block>children</Button>).container.firstChild
  ).toMatchSnapshot();
});
