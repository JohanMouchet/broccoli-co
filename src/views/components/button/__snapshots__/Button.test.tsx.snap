// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders <Button /> without crashing 1`] = `
<button
  class="button block"
  type="button"
>
  children
</button>
`;
